spyder/utils/programs.py

```diff
@@ -138,6 +138,19 @@ def alter_subprocess_kwargs_by_platform(**kwargs):
         # We "or" them together
         CONSOLE_CREATION_FLAGS |= CREATE_NO_WINDOW
         kwargs.setdefault('creationflags', CONSOLE_CREATION_FLAGS)
+
+        # ensure Windows subprocess environment has SYSTEMROOT, but variable
+        # is case insensitive
+        if ((kwargs.get('env') is not None) and
+            ('SYSTEMROOT' not in map(str.upper, kwargs['env'].keys()))):

```

Comment:
What is this doing 🙃 ?

Reply:
When running a subprocess, an 'env' keyword argument can be given which defines the environment variables in the subprocess shell environment. If the 'env' keyword argument is not supplied, the current shell environment is used. For situations where one wants a clean environment, simply providing `env={}` is all that is required. Except for windows. For Windows systems, `systemroot` must be in the environment variables; unfortunately, it's not case sensitive and multiple variations may exist, e.g. `SystemRoot` or `SYSTEMROOT`.

Reply:
Understood! Was not clear at first :-)